presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchClient.java

```diff
@@ -0,0 +1,391 @@
+                    .map(SchemaTableName::getSchemaName)
+                    .collect(toImmutableList());
+    }
+
+    public List<SchemaTableName> listTables(Optional<String> schemaName)
+    {
+        return tableDescriptions.getAllSchemaTableNames()
+                .stream()
+                .filter(schemaTableName -> !schemaName.isPresent() || schemaTableName.getSchemaName().equals(schemaName.get()))
+                .collect(toImmutableList());
+    }
+
+    private List<ColumnMetadata> loadColumns(ElasticsearchTableDescription table)
+    {
+        if (table.getColumns().isPresent()) {
+            return buildMetadata(table.getColumns().get());
+        }
+        List<ElasticsearchColumn> columns = buildColumns(table);
+        return buildMetadata(columns);
+    }
+
+    public List<ColumnMetadata> getColumnMetadata(ElasticsearchTableDescription tableDescription)
+    {
+        return tableColumnsCache.getUnchecked(tableDescription);
+    }
+
+    public ElasticsearchTableDescription getTable(String schemaName, String tableName)
+    {
+        requireNonNull(schemaName, "schemaName is null");
+        requireNonNull(tableName, "tableName is null");
+        ElasticsearchTableDescription table = tableDescriptions.get(new SchemaTableName(schemaName, tableName));
+        if (table == null) {
+            return null;
+        }
+        if (table.getColumns().isPresent()) {
+            return table;
+        }
+        List<ElasticsearchColumn> columns = buildColumns(table);
+        ElasticsearchTableDescription result = new ElasticsearchTableDescription(
+                table.getTableName(),
+                table.getSchemaName(),
+                table.getHost(),
+                table.getPort(),
+                table.getClusterName(),
+                table.getIndex(),
+                table.getIndexExactMatch(),
+                table.getType(),
+                Optional.of(columns));
+        return result;
+    }
+
+    public List<String> getIndices(ElasticsearchTableDescription tableDescription)
+    {
+        if (tableDescription.getIndexExactMatch()) {
+            return ImmutableList.of(tableDescription.getIndex());
+        }
+        TransportClient client = clients.get(tableDescription.getClusterName());
+        verify(client != null, "TransportClient is null");
+        String[] indices = client.admin()
+                .indices()
+                .getIndex(new GetIndexRequest())
+                .actionGet(timeout.toMillis())
+                .getIndices();
+        return Arrays.stream(indices)
+                    .filter(index -> index.startsWith(tableDescription.getIndex()))
+                    .collect(toImmutableList());
+    }
+
+    public ClusterSearchShardsResponse getSearchShards(String index, ElasticsearchTableDescription tableDescription)
+    {
+        TransportClient client = clients.get(tableDescription.getClusterName());
+        verify(client != null, "TransportClient is null");
```

Comment:
`verify(client != null, "client is null");`

Reply:
will fix